src/Serilog.Extras.TraceListener/Extras/TraceListener/SerilogTraceListener.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Copyright 2015 Serilog Contributors
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using Serilog.Core.Pipeline;
+using Serilog.Events;
+
+namespace Serilog.Extras.TraceListener
+{
+    /// <summary>
+    ///     TraceListener implementation that directs all output to Serilog.
+    /// </summary>
+    public class SerilogTraceListener : System.Diagnostics.TraceListener
+    {
+        const LogEventLevel FailLevel = LogEventLevel.Fatal;
+        const string MessageMessageTemplate = "{TraceMessage:l}";
+        const string FailMessageTemplate = "Fail: " + MessageMessageTemplate;
+        const LogEventLevel DefaultLogLevel = LogEventLevel.Debug;
+        ILogger logger;
+
+        /// <summary>
+        ///     Creates a SerilogTraceListener that uses the logger from `Serilog.Log`
+        /// </summary>
+        /// <remarks>
+        ///     This is needed because TraceListeners are often configured through XML
+        ///     where there would be no opportunity for constructor injection
+        /// </remarks>
+        public SerilogTraceListener() : this(Log.Logger)
+        {
+        }
+
+        /// <summary>
+        ///     Creates a SerilogTraceListener that uses the specified logger
+        /// </summary>
+        public SerilogTraceListener(ILogger logger)
+        {
+            this.logger = logger.ForContext<SerilogTraceListener>();
+        }
+
+        public override bool IsThreadSafe
+        {
+            get { return true; }
+        }
+
+        public override void Write(string message)
+        {
+            logger.Write(DefaultLogLevel, MessageMessageTemplate, message);
+        }
+
+        public override void Write(string message, string category)
+        {
+            logger.Write(DefaultLogLevel, "{Category:l}: " + MessageMessageTemplate, category, message);
```

Comment:
If no one cares about the repetition I am happy to change it.  To be honest, I initially did write out all the strings and once I had everything working I refactored it to how you see it now to avoid any possible concerns with duplication.

I personally had less concern with this one (because it is `const` like you mention), than `WriteEvent` a little further down where the message string is built conditionally based on the parameters.

I wasn't sure if doing things for performance reasons would be considered "premature optimization"